Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1014,6 +1057,80 @@ getStats (driverId, merchantId) date = do
            in Money $ GHCL.sum driverSelFares' + GHCL.sum customerExtFees' + GHCL.sum deadKmFare'
       }
 
+driverPhotoUploadHitsCountKey :: Id SP.Person -> Text
+driverPhotoUploadHitsCountKey driverId = "BPP:ProfilePhoto:verify:" <> getId driverId <> ":hitsCount"
+
+driverPhotoUpload :: (Id SP.Person, Id DM.Merchant) -> DriverPhotoUploadReq -> Flow APISuccess
+driverPhotoUpload (driverId, merchantId) DriverPhotoUploadReq {..} = do
+  checkSlidingWindowLimit (driverPhotoUploadHitsCountKey driverId)
+  person <- runInReplica $ QPerson.findById driverId >>= fromMaybeM (PersonNotFound (getId driverId))
+  encImage <- L.runIO $ base64Encode <$> BS.readFile image
+  imageExtension <- validateContentType
+  let req = IF.FaceValidationReq {file = encImage}
+  _ <- validateFaceImage merchantId req
+  filePath <- createFilePath (getId driverId) fileType imageExtension
+  transporterConfig <- CQTC.findByMerchantId (person.merchantId) >>= fromMaybeM (TransporterConfigNotFound (getId (person.merchantId)))
+  let fileUrl =
+        transporterConfig.mediaFileUrlPattern
+          & T.replace "<DOMAIN>" "driver-profile-picture"
+          & T.replace "<FILE_PATH>" filePath
+  result <- try @_ @SomeException $ S3.put (T.unpack filePath) encImage
+  case result of
+    Left err -> throwError $ InternalError ("S3 Upload Failed: " <> show err)
+    Right _ -> do
+      case person.faceImageId of
+        Just mediaFileId -> do
+          Esq.runTransaction $ do
+            QPerson.updateMediaId driverId Nothing
```

Comment:
Why are you deleting media here?

Reply:
We will not have any use for the previous image entries in mediafile table in case  the driver updates his profile photo.